build.gradle

```diff
@@ -261,6 +261,12 @@ subprojects { subproject ->
 		archives javadocJar
 	}
 
+	jar {
+		manifest {
+			attributes('Implementation-Version': version)
+		}
```

Comment:
I looked at the boot code to see how it puts the version in the banner.

I must admit that I didn't look to see whatever else boot puts in the manifest:

We should consider doing the same:

```
Manifest-Version: 1.0
Implementation-Title: Spring Boot
Implementation-Version: 1.3.3.RELEASE
Archiver-Version: Plexus Archiver
Built-By: bamboo
Specification-Vendor: Pivotal Software, Inc.
Specification-Title: Spring Boot
Implementation-Vendor-Id: org.springframework.boot
Created-By: Apache Maven 3.0.5
Build-Jdk: 1.8.0_60
Specification-Version: 1.3.3.RELEASE
Implementation-Vendor: Pivotal Software, Inc.
```

Perhaps not critical for M2, though.